topostats/tracing/skeletonize.py

```diff
@@ -2,78 +2,641 @@
+
+    Notes
+    -----
+    This code contains only the minimum viable product code as much of the other code
+    relating to pruning based on heights was unused. This also means that
+    should someone be upto the task, it is possible to include the heights when pruning.
     """
-    if method == "zhang":
-        return _skeletonize_zhang
-    if method == "lee":
-        return _skeletonize_lee
-    if method == "medial_axis":
-        return _skeletonize_medial_axis
-    if method == "thin":
-        return _skeletonize_thin
-    raise ValueError(method)
 
+    def __init__(self, image: np.ndarray, skeleton: np.ndarray) -> np.ndarray:
+        """Initialise the class.
+
+        Parameters
+        image: np.ndarray
+            The original data to help with branch removal.
+
+        skeleton np.ndarray
+            The skeleton to remove unwanted branches from.
+
+        Returns:
+            np.ndarray: _description_
+        """
+        self.image = image
+        self.skeleton = skeleton.copy()
+
+    def prune_all_skeletons(self) -> np.ndarray:
+        """Wrapper function to prune all skeletons by labling and iterating through
+        each one, binarising, then pruning, then adding up the single skeleton masks
+        to make a single mask.
+
+        Returns
+        -------
+        np.ndarray
+            A single mask with all pruned skeletons."""
+        pruned_skeleton_mask = np.zeros_like(self.skeleton)
+        for i in range(1, label(self.skeleton).max() + 1):
+            single_skeleton = self.skeleton.copy()
+            single_skeleton[single_skeleton != i] = 0
+            single_skeleton[single_skeleton == i] = 1
+            pruned_skeleton_mask += self._prune_single_skeleton(single_skeleton)
+        return pruned_skeleton_mask
+
+    def _prune_single_skeleton(self, single_skeleton: np.ndarray) -> np.ndarray:
+        """Function to remove the hanging branches from a single skeleton as this
+        function is an iterative process. These are a persistent problem in the
+        overall tracing process.
+
+        Parameters
+        ---------
+        single_skeleton: np.ndarray
+            A binary array containing a single skeleton.
+
+        Returns:
+        --------
+        np.ndarray
+            A binary mask of the single skeleton
+        """
+        pruning = True
+        while pruning:
+            number_of_branches = 0
+            coordinates = np.argwhere(single_skeleton == 1).tolist()
+
+            # The branches are typically short so if a branch is longer than
+            #  0.15 * total points, its assumed to be part of the real data
+            max_branch_length = int(len(coordinates) * 0.15)
+
+            # first check to find all the end coordinates in the trace
+            potential_branch_ends = self._findBranchEnds(coordinates)
+
+            # Now check if its a branch - and if it is delete it
+            for x_b, y_b in potential_branch_ends:
```

Comment:
Presumably `x_b` and `y_b` are co-ordinates of the branch. We can avoid `invalid-name` linting errors by instead using `x_branch` and `y_branch`.